app/controllers/admin_applications_controller.rb

```diff
@@ -0,0 +1,18 @@
+class AdminApplicationsController < ApplicationController
+  def index
+    @applications = Application.all
+  end
+  
+  def show
+    @application = Application.find(params[:id])
+  end
+
+  def update
+    pet = Pet.find(params[:pet_id])
+    application = Application.find(params[:id])
+    pet_application = PetApplication.where("pet_id = #{pet.id}", "application_id = #{application.id}")
```

Comment:
Per your question in your video: a slightly more MVC-friendly method here could be `find_by`. You could do: `pet_application = PetApplication.find_by(pet_id = pet.id, application_id = application.id)`
It's not wildly different than a where, but because where clauses are often use for more complex querying (i.e. things beyond exact value searches) we encourage students to remember `.where` as a model method clause. IMO this is not a huge MVC violation but I would swap it out with find_by and use key/value syntax to do exact value searching.